### PR TITLE
[GLUTEN-7307][VL] Update openssl version in velox setup for centos9

### PR DESCRIPTION
**Pull request by pratham76**
## What changes were proposed in this pull request?

The current velox [setup script](https://github.com/apache/incubator-gluten/blob/main/ep/build-velox/src/get_velox.sh) for centOS9 brings `opnssl-1.1.1s` which seems to be outdated all the current `spark-3.x` runtimes on centOS9 use `openssl-3.x` versions which causes the spark applications to fail due to this version mismatch.
To handle the above issue - upgraded the openssl version is the current setup.
Fixes: https://github.com/apache/incubator-gluten/issues/7307

## How was this patch tested?

manual tests

### CODE REVIEW

File: ep/build-velox/src/get_velox.sh [hunk 1]
```diff
@@ -107,7 +107,7 @@ function process_setup_centos9 {
   sed -i '/^.*dnf_install autoconf/a\  dnf_install libxml2-devel libgsasl-devel libuuid-devel' scripts/setup-centos9.sh
   
   ensure_pattern_matched 'install_gflags' scripts/setup-centos9.sh
-  sed -i '/^function install_gflags.*/i function install_openssl {\n  wget_and_untar https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_1_1_1s.tar.gz openssl \n  cd openssl \n  ./config no-shared && make depend && make && sudo make install \n  cd ..\n}\n'     scripts/setup-centos9.sh
+  sed -i '/^function install_gflags.*/i function install_openssl {\n  wget_and_untar https://github.com/openssl/openssl/releases/download/openssl-3.2.2/openssl-3.2.2.tar.gz openssl \n  cd openssl \n  ./config no-shared && make depend && make && sudo make install \n  cd ..\n}\n'     scripts/setup-centos9.sh
```
Comment (zhouyuan) [inline ep/build-velox/src/get_velox.sh:110]: should use `cd ${DEPENDENCY_DIR}/openssl` as recent Velox have this change?

Comment (pratham76) [reply ep/build-velox/src/get_velox.sh:110]: Velox doesn't seem to be bringing openssl 3.x  as a part of [centOS9 setup](https://github.com/oap-project/velox/blob/17ec2bb2d80f7db4093ca33546dbcb37d8ec9748/scripts/setup-centos9.sh).

Comment (majetideepak) [reply ep/build-velox/src/get_velox.sh:110]: @pratham76 `scripts/setup-centos9.sh` is coming from Velox.
In Velox, there was a recent change https://github.com/facebookincubator/velox/pull/10920 to use `${DEPENDENCY_DIR}` for dependencies. As @zhouyuan suggested, we need to use `cd ${DEPENDENCY_DIR}/openssl` when `wget_and_untar` is used.
Can you validate your change on the latest gluten main branch?

Comment (pratham76) [reply ep/build-velox/src/get_velox.sh:110]: got it. will update and validate.
This change does not seem to be added to gluten main branch yet - there are other dependencies being installed in current directory.

Comment (majetideepak) [reply ep/build-velox/src/get_velox.sh:110]: Interesting! Velox is updated daily and I do see this PR here https://github.com/oap-project/velox

Comment (pratham76) [reply ep/build-velox/src/get_velox.sh:110]: yes but gluten overrides those and adds new installs in this [script](https://github.com/apache/incubator-gluten/blob/main/ep/build-velox/src/get_velox.sh) which also has to be updated.

Comment (majetideepak) [reply ep/build-velox/src/get_velox.sh:110]: Then why is ${DEPENDENCY_DIR} not being used?

Comment (pratham76) [reply ep/build-velox/src/get_velox.sh]: Have updated the changes as requested and validated the build with manual tests.  Kindly review the changes. In the above comment i was noting that as gluten overrides the velox setup scripts using `sed` changes wrt install directories have to be added here too.

File: ep/build-velox/src/get_velox.sh [hunk 2]
```diff
@@ -107,7 +107,7 @@ function process_setup_centos9 {
   sed -i '/^.*dnf_install autoconf/a\  dnf_install libxml2-devel libgsasl-devel libuuid-devel' scripts/setup-centos9.sh
   
   ensure_pattern_matched 'install_gflags' scripts/setup-centos9.sh
-  sed -i '/^function install_gflags.*/i function install_openssl {\n  wget_and_untar https://github.com/openssl/openssl/archive/refs/tags/OpenSSL_1_1_1s.tar.gz openssl \n  cd openssl \n  ./config no-shared && make depend && make && sudo make install \n  cd ..\n}\n'     scripts/setup-centos9.sh
+  sed -i '/^function install_gflags.*/i function install_openssl {\n  wget_and_untar https://github.com/openssl/openssl/releases/download/openssl-3.2.2/openssl-3.2.2.tar.gz openssl \n  cd ${DEPENDENCY_DIR}/openssl \n  ./config no-shared && make depend && make && sudo make install \n}\n'     scripts/setup-centos9.sh
```
Comment (majetideepak) [inline ep/build-velox/src/get_velox.sh:110]: You have to use curly braces if you want to cd in a subshell and avoid `cd ../..` `( cd ${DEPENDENCY_DIR}/openssl \n  ./config no-shared && make depend && make && sudo make install )`

Comment (pratham76) [reply ep/build-velox/src/get_velox.sh]: updated. Will keep this in mind

Comment (pratham76) [reply ep/build-velox/src/get_velox.sh]: @majetideepak please review and lmk if any more changes are required. Thanks.

**Review by majetideepak — APPROVED** [review]
Thanks, @pratham76

**Comment by pratham76** [comment]
@zhouyuan Kindly review/approve the changes and inform if any more changes are required according to you. Thanks

**Review by zhouyuan — APPROVED** [review]
👍

**Pull request merged**